opencti-platform/opencti-front/src/private/components/analysis/malware_analyses/MalwareAnalysisCreation.tsx

```diff
@@ -0,0 +1,351 @@
+  product: string;
+  result_name: string;
+  version: string;
+  createdBy?: Option;
+  objectLabel: Option[];
+  submitted: Date;
+  result: string;
+  objectMarking: Option[],
+  externalReferences: Option[];
+  objectAssignee: Option[];
+  confidence: number
+}
+
+interface MalwareAnalysisFormProps {
+  updater: (store: RecordSourceSelectorProxy, key: string) => void
+  onReset?: () => void
+  onCompleted?: () => void
+  inputValue?: string
+  defaultCreatedBy?: Option
+  defaultMarkingDefinitions?: Option[]
+  defaultConfidence?: number
+}
+
+export const MalwareAnalysisCreationForm: FunctionComponent<MalwareAnalysisFormProps> = (
+  { updater,
+    onReset,
+    inputValue,
+    onCompleted,
+    defaultConfidence,
+    defaultCreatedBy,
+    defaultMarkingDefinitions },
+) => {
+  const classes = useStyles();
+  const { t } = useFormatter();
+  const basicShape = {
+    result_name: Yup.string().required(t('This field is required')),
+    product: Yup.string().required(t('This field is required')),
+    submitted: Yup.date().typeError(t('The value must be a datetime (yyyy-MM-dd hh:mm (a|p)m)')),
+    confidence: Yup.number(),
+  };
+  const malwareAnalysisValidator = useSchemaCreationValidation('Malware-Analysis', basicShape);
+  const initialValues: MalwareAnalysisAddInput = {
+    result_name: inputValue || '',
+    product: '',
+    version: '',
+    createdBy: defaultCreatedBy ?? '' as unknown as Option,
+    objectLabel: [],
+    submitted: dayStartDate(),
+    result: '',
+    objectMarking: defaultMarkingDefinitions ?? [],
+    externalReferences: [],
+    objectAssignee: [],
+    confidence: defaultConfidence ?? 75,
+  };
+  const [commit] = useMutation(malwareAnalysisMutation);
+  const onSubmit: FormikConfig<MalwareAnalysisAddInput>['onSubmit'] = (
+    values,
+    {
+      setSubmitting,
+      resetForm,
+      setErrors,
+    },
+  ) => {
+    const finalValues: MalwareAnalysisMutationVariables['input'] = {
+      result_name: values.result_name,
+      product: values.product,
+      version: values.version,
+      createdBy: values.createdBy?.value,
+      objectLabel: values.objectLabel.map((v) => v.value),
+      submitted: values.submitted,
+      result: values.result,
+      objectMarking: values.objectMarking.map(({ value }) => value),
+      confidence: parseInt(String(values.confidence), 10),
+      objectAssignee: values.objectAssignee.map(({ value }) => value),
+    };
+    commit({
+      variables: {
+        input: finalValues,
+      },
+      updater: (store) => {
+        if (updater) {
+          updater(store, 'malwareAnalysisAdd');
+        }
+      },
+      onError: (error: Error) => {
+        handleErrorInForm(error, setErrors);
+        setSubmitting(false);
+      },
+      onCompleted: () => {
+        setSubmitting(false);
+        resetForm();
+        if (onCompleted) {
+          onCompleted();
+        }
+      },
+    });
+  };
+
+  return <Formik<MalwareAnalysisAddInput>
```

Comment:
Is it okay to return without the " ( " ?

Reply:
I think so